sparkle/src/components/Citation.tsx

```diff
@@ -13,6 +13,8 @@ interface CitationProps {
   index?: ReactNode;
   isBlinking?: boolean;
   href?: string;
+  action?: React.ReactNode;
```

Comment:
Asking because this is unclear from just reading the code given the very limited typing directive which I think is not ideal for a design system?

Reply:
the existing `href` property does not let you add a "close" or "delete" action, so we added "action".

I think it makes the API of citation not the best but @Duncid thinks we should keep it as one component as much as we can (vs creating a new component).

We can also create a base component like "Card" and have two other components `ContentFragment` and `Citation` that use this one. LMK what you think @spolu .

Reply:
Note that there is an alternative that both conserve single component and API clarity by replacing href and action by a single "action" prop that is defined out of sparkle. However it requires some more work.

<Citation action={<Citation.Action onClick={} href whatEver/>} />

action can be a Citation.Action or a Button or whatever is requirered.
Citation.Action is just used to privide a "recommendation" of what action should be.
In our case a <IconButton tertiary>.

Reply:
I tried the `onClose` solution suggested on Slack, happy to revert if you are not happy with it @Duncid but just did that as a "branch prediction" in case you like it.

I updated the code, it still looks the same but I am adding a screenshot to help everybody with context:

<img width="1209" alt="Screen Shot 2023-11-16 at 12 17 07" src="https://github.com/dust-tt/dust/assets/358965/c301c858-fe07-4796-b7df-a6413f914196">